apps/api/src/app/plugins/bffCache.ts

```diff
@@ -0,0 +1,80 @@
+import fp from "fastify-plugin";
+import { CACHE_CONTROL_HEADER, getCache, getCacheControlHeaderValue, parseCacheControlHeaderValue, setCache } from "../../utils/cache";
+import { FastifyPluginCallback, FastifyReply, FastifyRequest } from "fastify";
+
+const HEADER_NAME = 'x-bff-cache'
+
+interface BffCacheOptions {
+  ttl?: number
+}
+
+export const bffCache: FastifyPluginCallback<BffCacheOptions> = (fastify, opts, next) => {
+  const { ttl } = opts
+  fastify.addHook('onRequest', async (request, reply) => {
+    // Cache only GET requests
+    if (request.method !== 'GET') {
+      return
+    }
+
+    let key = getKey(request)
+
+    const cacheItem = await getCache(key, fastify).catch(e => {
+      fastify.log.error(`Error getting key ${key} from cache`, e)
+      return null
+    })
+
+    if (cacheItem) {
+      const { item, ttl } = cacheItem
+      const ttlInSeconds = isNaN(ttl) ? undefined : Math.floor(ttl / 1000)
+
+      if (ttlInSeconds !== undefined) {
+        reply.header(CACHE_CONTROL_HEADER, getCacheControlHeaderValue(ttlInSeconds))
+        reply.header(HEADER_NAME, 'HIT')
+        reply.type('application/json')
+        reply.send(item)
+        return;
+      }
+    }
+
+    return
+  })
+
+  fastify.addHook('onSend', function (req, reply, payload, next) {
+    const isCacheHit = reply.getHeader(HEADER_NAME) === 'HIT'
+    const cacheTtl: number | undefined = getTtlFromResponse(reply, ttl)
+    const isStatus200 = reply.statusCode < 200 || reply.statusCode >= 300
```

Comment:
The variable name is inverted.
You are filtering out when it's not a 200.
Either update the name or the condition:

```suggestion
    const isNotStatus200 = reply.statusCode < 200 || reply.statusCode >= 300
```

```suggestion
    const isStatus200 = reply.statusCode >= 200 && reply.statusCode < 300
```